envoy/server/overload/overload_manager.h

```diff
@@ -101,5 +102,45 @@ class OverloadManager : public LoadShedPointProvider {
   virtual Event::ScaledRangeTimerManagerFactory scaledTimerFactory() PURE;
 };
 
+/**
+ * Implementation of OverloadManager that is never overloaded. Using this instead of the real
+ * OverloadManager keeps the interface accessible even when the proxy is overloaded.
+ */
+struct NullOverloadManager : public OverloadManager {
```

Comment:
Can you move this to the overload_manager_impl file to follow the standard convention within the codebase?